tests/backup/backup_helper.go

```diff
@@ -1473,12 +1473,17 @@ func GetAllRestoresNonAdminCtx(ctx context.Context) ([]string, error) {
 }
 
 // DeletePodWithLabelInNamespace kills pod with the given label in the given namespace
-func DeletePodWithLabelInNamespace(namespace string, label map[string]string) error {
+func DeletePodWithLabelInNamespace(namespace string, label map[string]string, ignoreLabel bool) error {
 	var pods *corev1.PodList
 	var err error
 	// TODO: Revisit this function and remove the below code if not needed
 	podList := func() (interface{}, bool, error) {
-		pods, err = core.Instance().GetPods(namespace, label)
+		if ignoreLabel {
```

Comment:
Instead of making this if check at multiple places could you calculate the podList based on ignoreLabel and use it later